qa/restart-upgrade/src/test/java/org/opensearch/neuralsearch/bwc/AbstractRestartUpgradeRestTestCase.java

```diff
@@ -8,14 +8,14 @@
 import java.util.Optional;
 import org.junit.Before;
 import org.opensearch.common.settings.Settings;
-import org.opensearch.neuralsearch.BaseNeuralSearchIT;
+import org.opensearch.neuralsearch.BaseSparseEncodingIT;
 import static org.opensearch.neuralsearch.TestUtils.CLIENT_TIMEOUT_VALUE;
 import static org.opensearch.neuralsearch.TestUtils.RESTART_UPGRADE_OLD_CLUSTER;
 import static org.opensearch.neuralsearch.TestUtils.BWC_VERSION;
 import static org.opensearch.neuralsearch.TestUtils.NEURAL_SEARCH_BWC_PREFIX;
 import org.opensearch.test.rest.OpenSearchRestTestCase;
 
-public abstract class AbstractRestartUpgradeRestTestCase extends BaseNeuralSearchIT {
+public abstract class AbstractRestartUpgradeRestTestCase extends BaseSparseEncodingIT {
```

Comment:
Why this is changed? I think such class hierarchy became misleading. It's better to keep extending BaseNeuralSearchIT and move methods from BaseSparseEncodingIT that you want to use to utility class

Reply:
`BaseNeuralSearchIT` should be at the bottom and be a base for everything. Something like:

AbstractRestartUpgradeRestTestCase  -> BaseNeuralSearchIT <- BaseSparseEncodingIT

regarding the utility class, I mean if you need to call some methods are in `BaseSparseEncodingIT` you can either move it to `BaseNeuralSearchIT` if that's applicable, or to utility class and call them from both `BaseSparseEncodingIT` (will need refactoring) and your class `AbstractRestartUpgradeRestTestCase`